src/frontends/pytorch/src/op/scaled_dot_product_attention.cpp

```diff
@@ -37,85 +20,34 @@ std::shared_ptr<ov::Node> translate_scaled_dot_product_attention_common(const No
-            auto mask_shape = context.mark_node(std::make_shared<v0::Concat>(OutputVector{tsl, ssl}, 0));
-            mask = context.mark_node(std::make_shared<v1::Broadcast>(minus_inf, mask_shape));
-            auto horizontal_range =
-                context.mark_node(std::make_shared<v4::Range>(zero_i, source_s_len, one_i, element::i32));
-            horizontal_range = context.mark_node(std::make_shared<v0::Unsqueeze>(horizontal_range, zero_i));
-            auto stop = context.mark_node(std::make_shared<v1::Add>(target_s_len, one_i));
-            auto vertical_range = context.mark_node(std::make_shared<v4::Range>(one_i, stop, one_i, element::i32));
-            vertical_range = context.mark_node(std::make_shared<v0::Unsqueeze>(vertical_range, one_i));
-            auto triu = context.mark_node(std::make_shared<v1::GreaterEqual>(horizontal_range, vertical_range));
-            atten_mask = context.mark_node(std::make_shared<v1::Select>(triu, mask, zero_f));
-        }
-        scaled_atten = context.mark_node(std::make_shared<v1::Add>(scaled_atten, atten_mask));
+    auto is_causal = context.input_is_none(5) ? false : context.const_input<bool>(5);
+    OutputVector inputs = {query, key, value};  // mandatory inputs
+
+    if (!context.input_is_none(3))
+        inputs.push_back(context.get_input(3));
+    else if (!context.input_is_none(6)) {
+        // need to fill a gap in inputs with scalar 0 to be able to pass one extra input after that
+        auto zero = op::v0::Constant::create(element::f32, Shape{}, {0});
+        inputs.push_back(context.mark_node(std::make_shared<v1::ConvertLike>(zero, query)));
     }
-    scaled_atten = context.mark_node(std::make_shared<v8::Softmax>(scaled_atten, -1));
-    return context.mark_node(std::make_shared<v0::MatMul>(scaled_atten, value));
-};
+    if (!context.input_is_none(6))
+        inputs.push_back(context.get_input(6));
+
+    return context.mark_node(std::make_shared<v13::ScaledDotProductAttention>(inputs, is_causal));
+}
 
 OutputVector translate_scaled_dot_product_attention(const NodeContext& context) {
     // aten::scaled_dot_product_attention(Tensor query, Tensor key, Tensor value, Tensor? attn_mask=None, float
-    // dropout_p=0., bool is_causal=False)
-    num_inputs_check(context, 6, 6);
+    // dropout_p=0., bool is_causal=False, float scale=None)
+    num_inputs_check(context, 6, 7);
     return {translate_scaled_dot_product_attention_common(context)};
 };
 
 OutputVector translate_scaled_dot_product_attention_fx(const NodeContext& context) {
     // aten::scaled_dot_product_attention(Tensor query, Tensor key, Tensor value, Tensor? attn_mask=None, float
-    // dropout_p=0., bool is_causal=False)
-    num_inputs_check(context, 3, 6);
+    // dropout_p=0., bool is_causal=False)  TODO: Scale parameter?
+    num_inputs_check(context, 3, 6);  // TODO: Set 7 instead of 6 if `scale` argument supported in FX.
```

Comment:
@cavusmustafa, please review this part: does FX have `scale`? Can I ran the test in "fx-mode"?

Reply:
@cavusmustafa, @suryasidd, @ynimmaga, I ran all scaled_dot_product_attention tests with PYTORCH_TRACING_MODE=TORCHFX. All tests passed but it gave me a warning ```UserWarning: changing options to `torch.compile()` may require calling `torch._dynamo.reset()` to take effect```. Does it mean that all tests are passing with `torch.compile`? Does it mean that they are passing with `fx` backend? As I haven't changed the number of possible inputs in the `fx` part when introducing a new `scale` input, the test shouldn't pass, so I'm not sure that the `fx` part is covered. Could you help to test it?